src/toxcore/dht/dht_node.rs

```diff
@@ -31,9 +31,9 @@ If both node is Good node, then we compare PK's distance.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct DhtNode {
     /// Socket addr of node for IPv4.
-    pub saddr_v4: Option<SocketAddrV4>,
+    pub saddr_v4: Option<SocketAddr>,
     /// Socket addr of node for IPv6.
-    pub saddr_v6: Option<SocketAddrV6>,
+    pub saddr_v6: Option<SocketAddr>,
```

Comment:
Why change this to `SocketAddr`?

Reply:
Which means more opportunities to make a mistake. Isn't it wrong if `saddr_v4` stores IPv6 address or `saddr_v6` stores IPv4 address? I'd prefer to leave it as is but if you need these addresses as `SocketAddr` then just add two methods to `DhtNode` that converts them to `SocketAddr`.

Reply:
Ah. I got it. It is worse than before.